### ISSUE TITLE
The inverted boolean-check introduces subtle mistakes

### ISSUE THREAD

**Issue by jroith** [SonarSource/eslint-plugin-sonarjs#211]
**I want to report a bug.**

**Reproducer**

original:

```javascript
!(parseInt(c, 10) >= 0)
```

rewritten to:

```javascript
parseInt(c, 10) < 0
```

The original returns true if c is a numeric string like "42" and false for other strings like "abc". This now breaks and always returns false.

**Expected behavior**

Should return true if called with a non-numeric string, but returns false.

**eslint-plugin-sonarjs version:** 0.7.0<br>
**eslint version:** 7.25.0<br>
**Node.js version:** 12.0

**Rule key:** sonarjs/no-inverted-boolean-check

**Comment by vilchik-elena** [comment]
Let's remove the rule from default configuration and update description mentioning that rule should be activated only if NaN is not expected.
